### PR TITLE
Added holidays for Chile

### PR DESCRIPTION
**Pull request by diegoibanezr**

### CODE REVIEW

**Comment by nexxai** [comment]
You'll need to provide the snapshot to have the test check against.  You can check some of the other PRs to see how it should be formatted.

File: src/Countries/Chile.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+use Carbon\CarbonImmutable;
+
+class Chile extends Country
+{
+    public function countryCode(): string
+    {
+        return 'cl';
+    }
+
+    /** @return array<string, CarbonImmutable> */
```
Comment (Nielsvanpach) [inline src/Countries/Chile.php:14]: can be removed

File: src/Countries/Chile.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            'Día de Todos los Santos' => '11-01',
+            'Inmaculada Concepción' => '12-08',
+            'Navidad' => '12-25',
+        ], $this->variableHolidays($year));
+    }
+
+    /** @return array<string, CarbonImmutable> */
+    protected function variableHolidays(int $year): array
+    {
+        $easter = CarbonImmutable::createFromTimestamp(easter_date($year))->setTimezone('America/Santiago');
```
Comment (Nielsvanpach) [inline src/Countries/Chile.php:38]: Could you use `$this->easter($year)` instead? Available after rebasing from this repo's main branch.

File: tests/Countries/ChileTest.php [hunk 3]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace Spatie\Holidays\Tests\Countries;
+
+use Carbon\CarbonImmutable;
+use Spatie\Holidays\Holidays;
+
+it('can calculate Chile holidays', function () {
```
Comment (Nielsvanpach) [inline tests/Countries/ChileTest.php:8]: Could you run the tests locally too and verify if the output is correct?

**Review by Nielsvanpach — COMMENTED** [review]
Thanks!

File: tests/Countries/ChileTest.php (hunk 3 above)
Comment (Nielsvanpach) [reply tests/Countries/ChileTest.php:8]: No worries. :-)

Make sure your fork is synced with this repo: 
https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork

Git rebase integrates changes from the main branch into your branch by applying your branch's commits on top of the latest main branch commits, resulting in a linear history. How to do this, might be best to google yourself.

Running the tests locally should generate a snapshot file. You'll have to commit that file too.

**Comment by Nielsvanpach** [comment]
Closing as another PR for Chile has been merged.

**Pull request closed without merging**